syft/frameworks/torch/hook.py

```diff
@@ -273,19 +275,21 @@ def overloaded_attr(_self, *args, **kwargs):
                 # Store this utility function in the registry
                 hook_self.args_hook_for_overloaded_attr[attr] = hook_args_function
 
-            try:
+            # has_child = hasattr(_self, "child")
+            # if(has_child):
+            #     has_pointer_child = isinstance(_self.child, syft.frameworks.torch.tensors.PointerTensor)
+            # else:
+            #     has_pointer_child = False
+
+            # TODO: change if statement to "if has_pointer_child"
+
+            if not isinstance(_self, syft.frameworks.torch.tensors.PointerTensor):
                 # Transform the args
-                try:
-                    # Load the utility function to transform the args
-                    hook_args = hook_self.args_hook_for_overloaded_attr[attr]
-                    # Try running it
-                    new_self, new_args = hook_args((_self, args))
-                except IndexError:  # Update the function in cas of an error
```

Comment:
There is a subtleness here: some functions accept different signatures (especially something like f(*args), think of print) but the `hook_args` function is designed to work for a single signature: so it can break (with low probabilty) in which case we need to update it with the new signature conveyed by `args` using `build_hook_args_function`. Or we should store all versions instead of updating.

Reply:
It seems like hook_Self.args_hook_for_overloaded_attr is being checked on line 276 - thus there's no need to do it twice, correct?

Reply:
ok - after further discussion - i added this try/except back for now as I understand now why it's used. I also added a unit test showing what breaks if you remove it.